hooks/useUpdatePoolInstances/index.ts

```diff
@@ -125,8 +126,8 @@ export const useUpdatePoolInstances = (): void => {
     }, [provider, poolsInitialized]);
 
     // update token balances and approvals when address changes
-    useEffect(() => {
-        if (provider && account && poolsInitialized) {
+    useMemo(() => {
```

Comment:
why switch to useMemo here?

Reply:
https://stackoverflow.com/questions/56028913/usememo-vs-useeffect-usestate